splight_lib/models/hub_solution.py

```diff
@@ -23,27 +36,72 @@ class HubSolution(SplightDatabaseBaseModel):
     tags: Optional[List[str]] = Field(default=None)
     privacy_policy: PrivacyPolicy = PrivacyPolicy.PUBLIC
 
-    main_file: Optional[FilePath] = Field(default=None, exclude=True)
-    variables_file: Optional[FilePath] = Field(default=None, exclude=True)
-    values_file: Optional[FilePath] = Field(default=None, exclude=True)
-    readme_file: Optional[FilePath] = Field(default=None, exclude=True)
-
-    def save(self) -> None:
-        if not all(
-            [
-                self.main_file,
-                self.variables_file,
-                self.values_file,
-                self.readme_file,
-            ]
-        ):
-            raise ValueError("Missing one of the required files")
-        return super().save()
-
-    def _get_model_files_dict(self) -> Optional[Dict]:
-        return {
-            "main_file": self.main_file,
-            "variables_file": self.variables_file,
-            "values_file": self.values_file,
-            "readme_file": self.readme_file,
-        }
+    config: List[InputParameter] = []
+    resources: List[InputAsset] = []
+
+    @classmethod
+    def upload(cls, path: FilePath):
+        db_client = cls._SplightDatabaseBaseModel__get_database_client()
```

Comment:
Could you let me know if you meant the following? What is this otherwise?
```suggestion
        db_client = cls._SplightDatabaseBaseModel.__get_database_client()
```

Reply:
Oh interesting! I didn't know we could do this in Python